packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -1477,7 +1485,11 @@ export class BranchIfNotZero extends Op {
     const last = this.assertBigInt(stack.pop(), this.line);
 
     if (last !== 0n) {
-      this.interpreter.jumpForward(this.label, this.line);
+      if (this.interpreter.tealVersion <= 3) {
+        this.interpreter.jumpForward(this.label, this.line);
+      } else {
+        this.interpreter.jumpToLabel(this.label, this.line);
+      }
```

Comment:
We have this if else 3 times. Shall we move this check to `jumpForward`/ `jumpToLabel` ? Eg.
```js
function jumpForward(..) {
 if (this.tealVersion > 3) { this.jumpToLabel(label, line); }
 ..
}
```
OR, add move this to say `interpreter.jump(label, line)` ?

Reply:
this was discussed in friday call, now we will be adding different class for these type of opcodes, for ex: b3, b4. b4 will inherit b3 if needed.

Reply:
How about having the name convention: `Bv3`, `Bv4` ...